src/services/SlotService.test.ts

```diff
@@ -0,0 +1,58 @@
+import sinon from "sinon";
+import { stubInterface } from "ts-sinon";
+import Slot from "../entities/Slot";
+import IRepository from "../types/IRepository";
+import getSlotMock from "../mocks/SlotMock";
+import SlotService from "./SlotService";
+
+describe(
+  "SlotService", () => {
+    const sandbox = sinon.createSandbox();
+
+    afterEach(() => sandbox.restore());
+
+    it(
+      "create should generate a new slot when called and return its data", async () => {
+        // Given
+        const slotMock = getSlotMock();
+        const fakeRepo = stubInterface<IRepository<Slot>>();
+        const userInput = {
+          isDisabled: true,
+        };
+
+        // When
+        fakeRepo.save.resolves(slotMock);
+        sandbox.replace(
+          SlotService.prototype, "getRepository", () => fakeRepo,
+        );
+
+        // Then
+        const res = await new SlotService().create(userInput);
+
+        expect(res).toEqual(slotMock);
+        expect(fakeRepo.save.calledWithExactly(userInput)).toBeTruthy();
+      },
+    );
+
+    it(
+      "create fails due to missing parameters", async () => {
+        // Given
+        const fakeRepo = stubInterface<IRepository<Slot>>();
+
+        // When
+        fakeRepo.save.throws();
+        sandbox.replace(
+          SlotService.prototype, "getRepository", () => fakeRepo,
+        );
+
+        // Then
+        await expect(new SlotService().create({
+          isDisabled: undefined,
```

Comment:
It shouldn't fail because of this, if we have parameters that have a default value such as this one, we can just let it through. The problem is when you have something like a required email.